File: src/net/intellij/plugins/sexyeditor/SexyEditorState.java
package net.intellij.plugins.sexyeditor;

import com.intellij.openapi.components.PersistentStateComponent;
import com.intellij.openapi.components.ServiceManager;
import com.intellij.openapi.components.State;
import com.intellij.openapi.components.Storage;
import com.intellij.util.xmlb.XmlSerializerUtil;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.List;

@State(
	name = SexyEditor.COMPONENT_NAME,
	storages = {@Storage(id = "sexyeditor", file = "sexyeditor.xml")})
public class SexyEditorState implements PersistentStateComponent<SexyEditorState> {

	public List<BackgroundConfiguration> configs = new ArrayList<>();

	@Nullable
	public static SexyEditorState getInstance() {
		return ServiceManager.getService(SexyEditorState.class);
	}

	/**
	 * Returns plugin state.
	 */
	@Override
	public SexyEditorState getState() {
		return this;
	}

	/**
	 * Loads state from configuration file.
	 */
	@Override
	public void loadState(SexyEditorState state) {
		XmlSerializerUtil.copyBean(state, this);
	}

}

File: src/main/kotlin/com/github/igr/sexyeditor/config/SexyEditorState.kt
package com.github.igr.sexyeditor.config

import com.github.igr.sexyeditor.COMPONENT_NAME
import com.intellij.openapi.application.ApplicationManager
import com.intellij.openapi.components.PersistentStateComponent
import com.intellij.openapi.components.State
import com.intellij.openapi.components.Storage
import com.intellij.util.xmlb.XmlSerializerUtil

@State(
    name = COMPONENT_NAME,
    storages = [Storage("sexy-editor-3.xml")]
)
class SexyEditorState : PersistentStateComponent<SexyEditorState> {

    var configs: MutableList<BackgroundConfiguration> = mutableListOf()

    companion object {
        fun get(): SexyEditorState {
            return ApplicationManager.getApplication().getService(SexyEditorState::class.java)
        }
    }

    /**
     * Returns plugin state.
     */
    override fun getState(): SexyEditorState {
        return this
    }

    /**
     * Loads state from configuration file.
     */
    override fun loadState(state: SexyEditorState) {
        XmlSerializerUtil.copyBean(state, this)
    }
}